lib/api/model/initial_snapshot.dart

```diff
@@ -102,3 +105,98 @@ class RecentDmConversation {
 
   Map<String, dynamic> toJson() => _$RecentDmConversationToJson(this);
 }
+
+/// The `unread_msgs` snapshot.
+///
+/// For docs, search for "unread_msgs:"
+/// in <https://zulip.com/api/register-queue>.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class UnreadMessagesSnapshot {
+  final int count;
+
+  @JsonKey(name: 'pms')
+  final List<UnreadDmSnapshot> dms;
+
+  final List<UnreadStreamSnapshot> streams;
+  final List<UnreadHuddleSnapshot> huddles;
+  final List<int> mentions;
+  final bool oldUnreadsMissing;
+
+  UnreadMessagesSnapshot({
+    required this.count,
+    required this.dms,
+    required this.streams,
+    required this.huddles,
+    required this.mentions,
+    required this.oldUnreadsMissing,
+  });
+
+  factory UnreadMessagesSnapshot.fromJson(Map<String, dynamic> json) =>
+    _$UnreadMessagesSnapshotFromJson(json);
+
+  Map<String, dynamic> toJson() => _$UnreadMessagesSnapshotToJson(this);
+}
+
+/// An item in `unread_msgs.pms`.
+///
+/// For docs, search for "unread_msgs:" and see "pms:"
+/// in <https://zulip.com/api/register-queue>.
```

Comment:
nit:
```suggestion
/// An item in [UnreadMessagesSnapshot.dms].
```

(and similarly the other two below)

I think once you search for "unread_msgs:" and are looking at that section, it's easy to find the individual subsections for these pieces of it. Also if you're looking at any one of them it's quite likely you're looking at other parts of `unread_msgs` anyway.

Conversely it's nice to draw the links between these types explicitly, and to keep them each compact the better to have several of them on screen at the same time.